client/go/cmd/login.go

```diff
@@ -18,7 +18,15 @@ var loginCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
-		a, err := auth0.GetAuth0(cfg.AuthConfigPath(), getSystemName(), getApiURL())
+		targetType, err := getTargetType()
+		if err != nil {
+			return err
+		}
+		system, err := getSystem(targetType)
+		if err != nil {
+			return err
+		}
+		a, err := auth0.GetAuth0(cfg.AuthConfigPath(), system.Name, system.URL)
```

Comment:
I think it's worth to avoid coupling the `auth0`and `vespa` packages too much. In general, I think the `auth0` package could use some refactoring, but not in this PR.